packages/components/src/components/loaders/Spinner/Spinner.tsx

```diff
@@ -1,51 +1,78 @@
-import React from 'react';
+import Lottie from 'lottie-react';
+import React, { useMemo, useState } from 'react';
 import styled from 'styled-components';
-import { animations } from '../../../config';
+import animationStart from './animationData/refresh-spinner-start.json';
+import animationMiddle from './animationData/refresh-spinner-middle.json';
+import animationEnd from './animationData/refresh-spinner-end-success.json';
+import animationWarn from './animationData/refresh-spinner-end-warning.json';
+
+const StyledLottie = styled(Lottie)<Pick<FluidSpinnerProps, 'size' | 'isGrey'>>`
+    width: ${({ size }) => `${size}px`};
+    height: ${({ size }) => `${size}px`};
+    filter: ${({ isGrey }) => (isGrey ? 'grayscale(1) opacity(0.6)' : 'none')};
+`;
 
 export interface FluidSpinnerProps {
     size?: number;
-    strokeWidth?: number;
-    color?: string;
+    isGrey?: boolean;
+    isWithSpeadup?: boolean;
+    hasFinished?: boolean;
+    hasError?: boolean;
     className?: string;
 }
 
-const Wrapper = styled.div<FluidSpinnerProps>`
-    /* https://loading.io/css/ */
-    position: relative;
-    width: ${({ size }) => `${size}px`}; /* change to 1em to scale based on used font-size */
-    height: ${({ size }) => `${size}px`}; /* change to 1em to scale based on used font-size */
-
-    div {
-        position: absolute;
-        box-sizing: border-box;
-        width: ${({ size }) => `${size}px`}; /* change to 1em to scale based on used font-size */
-        height: ${({ size }) => `${size}px`}; /* change to 1em to scale based on used font-size */
-        border: ${({ strokeWidth }) => `${strokeWidth}px`} solid transparent; /* change to 0.1em to scale based on used font-size */
-        border-radius: 50%;
-        animation: ${animations.SPIN} 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
-        border-color: #fff transparent transparent transparent;
-        border-top-color: ${({ color }) => color || 'inherit'};
-        will-change: transform;
-    }
+export const Spinner = ({
+    size = 100,
+    isGrey = true,
+    isWithSpeadup,
+    hasFinished,
+    hasError,
+    className,
+}: FluidSpinnerProps) => {
+    const [hasStarted, setHasStarted] = useState(false);
+    const [hasFinishedRotation, setHasFinishedRotation] = useState(false);
 
-    div:nth-child(1) {
-        animation-delay: -0.45s;
-    }
+    const commonProps = {
+        size,
+        isGrey,
+        className,
+    };
 
-    div:nth-child(2) {
-        animation-delay: -0.3s;
+    const onLoopComplete = useMemo(
+        () => () => {
+            if (!hasFinished || !hasError) {
+                return;
+            }
+
+            setHasFinishedRotation(true);
+        },
+        [hasFinished, hasError],
+    );
```

Comment:
You could `useCallback`. However, I am not sure if the memoization is helping anything here.

Reply:
b3f3c48bf5